app/v1/controllers/requestParsers/validators/validations/RetrievePensionChargesValidator.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators.validations
+
+import config.AppConfig
+import javax.inject.Inject
+import v1.controllers.requestParsers.validators.Validator
+import v1.models.errors.{MtdError, TaxYearFormatError}
+import v1.models.requestData.RetrievePensionChargesRawData
+
+class RetrievePensionChargesValidator @Inject()(appConfig: AppConfig) extends Validator[RetrievePensionChargesRawData]{
+
+  private val validationSet = List(parameterFormatValidation)
+
+  private def parameterFormatValidation: RetrievePensionChargesRawData => List[List[MtdError]] = { data =>
```

Comment:
I don't think you need the vals as you can just use a similar format to how ListBFLossesValidator does it's validation. The .distinct should take care of duplicate error so the if statement shouldn't be needed I don't think.

Reply:
The reason it was done like this in the last one was so that we didn't get MinTaxYearValidation errors back if the taxYear format was wrong.

Something Charlie spotted last week.